File: ReglaPeso.py
from ReglaPrecio import ReglaPrecio


class ReglaPeso(ReglaPrecio):

    def calcular_total(self, cantidad, precio):
        peso = cantidad * 1000
        return precio * peso
    
    def es_aplicable(self, sku):
        return sku.startswith("WE")

File: Backend/model/ReglaPeso.ts
import ReglaPrecio from "./ReglaPrecio";

export default class ReglaPeso implements ReglaPrecio{

    //cantidad viene dada en kilogramos
    calcularTotal(cantidad: number, precio: number): number {
        const gramos = cantidad*1000;
        return gramos * precio;
    }

}